diesel_infer_schema/src/inference.rs

```diff
@@ -143,14 +143,20 @@ pub fn load_foreign_key_constraints(database_url: &str, schema_name: Option<&str
 {
     let connection = try!(establish_connection(database_url));
 
-    match connection {
+    let mut constraints = match connection {
         #[cfg(feature = "sqlite")]
         InferConnection::Sqlite(c) => ::sqlite::load_foreign_key_constraints(&c, schema_name),
         #[cfg(feature = "postgres")]
         InferConnection::Pg(c) => ::information_schema::load_foreign_key_constraints(&c, schema_name).map_err(Into::into),
         #[cfg(feature = "mysql")]
         InferConnection::Mysql(c) => ::mysql::load_foreign_key_constraints(&c, schema_name).map_err(Into::into),
+    };
+
+    if let Ok(ref mut c) = constraints.as_mut() {
```

Comment:
That's pretty… noisy code with many ref/mut/as_mut thingies. Do you think `let constraints = constraints.map(|mut cs| { cs.sort(); cs });` is better?

Reply:
Good point! Just pushed a cleanup for that. No need for an assignment, that can be the terminating expression for the function.

(related: i simply cannot wait for pattern ergonomics improvements!)